lib/pure/os.nim

```diff
@@ -3310,3 +3310,61 @@ func isValidFilename*(filename: string, maxLen = 259.Positive): bool {.since: (1
     find(f.name, invalidFilenameChars) != -1): return false
   for invalid in invalidFilenames:
     if cmpIgnoreCase(f.name, invalid) == 0: return false
+
+when defined(windows):
+  const ComputerNamePhysicalNetBIOS = 4
+  const ComputerNamePhysicalDnsHostname = 5
+
+proc getHostName*: string {.noWeirdTarget.} =
+  ## Gets host name. Available in Windows and POSIX.
+  var success: bool
+  when defined(posix):
+    const maxHostNameLen = 256
+    result = newString(maxHostNameLen)
+    success = posix.gethostname(result.cstring, maxHostNameLen) == 0
+  elif defined(windows):
+    var maxHostNameLen: DWORD = 256
+    result = newString(maxHostNameLen)
+    proc getComputerNameExA(nameType: cint, name: cstring, len: PDWORD): bool
+      {.stdcall, dynlib: "kernel32", importc: "GetComputerNameExA", sideEffect.}
+    success = getComputerNameExA(ComputerNamePhysicalDnsHostname,
+                                 result.cstring, maxHostNameLen.addr):
+  else:
+    doAssert false, "getHostName failed: OS is not supported"
+  if not success:
+    raiseOSError(osLastError())
+  let x = len(cstring(result))
+  result.setLen(x)
+
+proc setHostName*(name: string) {.noWeirdTarget.} =
+  ## Sets host name.
+  ## Available in Windows, Linux (including Android), OS X, BSD, Haiku, and QNX
+  ## distributions.
+  ## Requires root priviledges on POSIX and administrator privileges on
+  ## Windows.
+  ## On Windows name changes made by this function call do not take effect
+  ## until the user restarts the computer.
```

Comment:
add link to https://docs.microsoft.com/en-us/windows/win32/sysinfo/computer-names and explain that on windows, it sets both ComputerNamePhysicalNetBIOS and ComputerNamePhysicalDnsHostname